### DISCUSSION TITLE
Command Syntaxis

### DISCUSSION THREAD

**Discussion by rtek1000** [arendst/Tasmota#14291] [Q&A] [answered]
It might be good to standardize the commands so that new users will find it easier to use the application.

For example:

```
IPAddress<x>
IPAddress1 = 192.168.1.123
```

```
NtpServer<x>
NtpServer 200.160.0.8
```

Note that just here we already have a de-standardization, after **IPAddress** comes the index without spacing. But for **NtpServer** does not accept command without spacing.

Also, it got confusing to use the equal sign (=) for **IPAddress** and for **NtpServer** it doesn't need the equal sign (=) but it does need a space.

To help circumvent the lack of standardization, an example of usage could be placed in each command statement, as was done with **IPAddress**

**Comment by ascillato** [comment, marked as answer]
Hi, 

Sorry. There is no command in Tasmota that allows the = sign. In the docs at commands, there is no example with =. The = is not Tasmota syntax.

All commands are standard in the form:

`COMMAND<INDEX> DATA`

It has only one SPACE between `INDEX` and `DATA`

The `IPAddress1 = 192.168.1.123` is **_not_** a valid Tasmota syntax. Typing that will make Tasmota to ignore your command and just show you the current value for `IPAddress1`. The only supported syntax is:

```haskell
IPAddress1 192.168.1.123
IPAddress2 192.168.1.1
NtpServer1 200.160.0.8
NtpServer2 200.160.0.9
```

If you type `NtpServer 200.160.0.8`, Tasmota will interpret that as `NtpServer1 200.160.0.8`. Omitting the index will be defaulted to 1.

**Reply by barbudor** [reply]
@ascillato There are some docs which have already misleading users to believe there should be an equal signal
IPAddress is one of those example:
![image](https://user-images.githubusercontent.com/2996042/148177663-bf642c89-d25f-432a-8533-c9539c1c6b23.png)

@rtk1000 With a very few exceptions, command syntax is very standardized. So it's mainly a problem of documentation.
- Developpers do not like to write docs (code is the doc), so sometime new features don't get properly documented
- Occasional collaborator may push a new feature without following the "standard". It may be because he did it quickly originally as a POC but then once the core feature is working, he doesn't take the time to align it. Once the feature is established and in-use, breaking changes are generally not welcomed.

The doc is open source like the code. You just have to click on the pencil icon on the top right to suggest a change. Any improvement is welcomed. It's just a matter of having the will to take time for the community.

**Reply by ascillato** [reply]
Docs fixed. The = symbol there was used as an English word (yes, it is very confusing) and not as an example. Now it is removed and the explanation is done all in words. These type of changes to the docs are very good for improving its readability. Thanks for pointing this out.
